System/Primitive.cpp

```diff
@@ -158,6 +158,102 @@ namespace RTE {
 		}
 	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void RoundBoxPrimitive::Draw(BITMAP *drawScreen, Vector targetPos) {
+		if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) {
+			Vector drawStart = m_StartPos - targetPos;
+			Vector drawEnd = m_EndPos - targetPos;
+
+			// Top left corner
+			arc(drawScreen, drawStart.m_X + m_CornerRadius, drawEnd.m_Y + m_CornerRadius, itofix(64), itofix(128), m_CornerRadius, m_Color);
```

Comment:
Am I reading this right? If this is the top left corner, why does it draw to the end position's Y? Is that not the bottom corner? Or is it cause corner radius deals with that somehow???